### DISCUSSION TITLE
RLS not workng..

### DISCUSSION THREAD

**Discussion by JulianSwales** [supabase/supabase#8054] [Questions] [answered]
I'm writing an application using Flutter.. I am attempting to enable RLS, however it's behaving oddly and I'm not sure where to look.. Whilst this is a single example below all my requests fail in the same manner. 
If RLS is disabled, the following call works, returns records: 

final response = await supabase
        .from(tableName())
        .select()
        .gte('booking_date', currentDate).lt('booking_date', currentDate.add(Duration(days: 365)))
        .order('booking_date', ascending: true)
        .execute();

If I enable RLS with no polices this call returns 0 records which I expect. 
If I then add a policy: 
<img width="1155" alt="Screen Shot 2022-08-02 at 11 24 04 AM" src="https://user-images.githubusercontent.com/66487981/182436426-6b479a6a-8a8c-4d0c-aa6b-d07bd5248d2c.png">

This works and returns records.. However if I change the policy to authenticated users, as below, it returns 0 rows.
<img width="1136" alt="Screen Shot 2022-08-02 at 11 26 17 AM" src="https://user-images.githubusercontent.com/66487981/182436745-fa5fd06f-e82f-4518-9f68-7ab0242c57ef.png">
However change authenticated to anon and it works. 

It's behaving as if the user isn't authenticated, but I do the logon process and I get back the authentication and the token. 
How can I tell if Supabase thinks I'm authenticated?  Why does Supabase not think I'm authenticated? 
I find debugging RLS very difficult since there's way to simulate a user.. 
Thanks.

**Comment by GaryAustin1** [comment]
It is not an RLS.  With anon working that means the anon key is being sent in your API request instead of a user JWT.  For some reason the user jwt is not set in the client at the time of the call.

**Comment by JulianSwales** [comment, marked as answer]
The Anon was the key from the answer above. I discovered I was initializing Supabase twice, so it was authenticating correctly, but then the queries were being run as anonymous from the second initialization. Once I removed the second initialization, everything started to work correctly.
